src/migration-scripts/interfaces/16-to-17

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+
+# Command line migration of port mirroring
+# https://phabricator.vyos.net/T3089
+
+import sys
+from vyos.configtree import ConfigTree
+
+if __name__ == '__main__':
+    if (len(sys.argv) < 1):
+        print("Must specify file name!")
+        sys.exit(1)
+
+    file_name = sys.argv[1]
+
+    with open(file_name, 'r') as f:
+        config_file = f.read()
+
+    config = ConfigTree(config_file)
+    base = ['interfaces', 'ethernet']
+    if not config.exists(base):
+        # Nothing to do
+        sys.exit(0)
+    else:
+        for interface in config.list_nodes(base):
+            mirror_old_base = base + [interface, 'mirror']
+            if config.exists(mirror_old_base):
+                for mirror_int in config.list_nodes(mirror_old_base):
+                    mirror_old = mirror_old_base + [mirror_int]
+                    if config.exists(mirror_old):
+                        config.set(mirror_old + ['direction','ingress'])
+                        # format as tag node to avoid loading problems
+                        config.set_tag(mirror_old_base)
+        try:
+            with open(file_name, 'w') as f:
+                f.write(config.to_string())
+        except OSError as e:
+            print("Failed to save the modified config: {}".format(e))
+            sys.exit(1)
```

Comment:
The migrator should look like this:

```
#!/usr/bin/env python3
#
# Copyright (C) 2020 VyOS maintainers and contributors
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License version 2 or later as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

# Command line migration of port mirroring
# https://phabricator.vyos.net/T3089

import sys
from vyos.configtree import ConfigTree

if __name__ == '__main__':
    if (len(sys.argv) < 1):
        print("Must specify file name!")
        sys.exit(1)

    file_name = sys.argv[1]

    with open(file_name, 'r') as f:
        config_file = f.read()

    config = ConfigTree(config_file)
    base = ['interfaces', 'ethernet']
    if not config.exists(base):
        # Nothing to do
        sys.exit(0)

    for interface in config.list_nodes(base):
        mirror_old_base = base + [interface, 'mirror']
        if config.exists(mirror_old_base):
            for mirror_int in config.list_nodes(mirror_old_base):
                mirror_old = mirror_old_base + [mirror_int]
                if config.exists(mirror_old):
                    config.set(mirror_old + ['direction','ingress'])
                    # format as tag node to avoid loading problems
                    config.set_tag(mirror_old_base)
    try:
        with open(file_name, 'w') as f:
            f.write(config.to_string())
    except OSError as e:
        print("Failed to save the modified config: {}".format(e))
        sys.exit(1)
```